### PR TITLE
Sphinx-based documentation infrastructure

### PR DESCRIPTION
**Pull request by densh**
This PR introduces basic infrastructure for documentation based on [Sphinx](http://www.sphinx-doc.org/en/stable/).

We're likely to host it through readthedocs.io or github pages on docs.scala-native.org subdomain.

To build docs use `make html` in `docs/` folder after installing sphinx via `pip install sphinx sphinx-autobuild`.

Review by @andreaTP @frgomes

### CODE REVIEW

File: docs/user/lang.rst [hunk 1]
```diff
@@ -0,0 +1,9 @@
+.. _lang:
+
+Language semantics
+==================
+
+In general, the semantics of the Scala.js language are the same as Scala on the
```
Comment (pbatko) [inline docs/user/lang.rst]: `s/Scala.js/scala-native/`?

Comment (densh) [inline docs/user/lang.rst]: Yep!

File: docs/user/sbt.rst [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+and now you can write your first application in ``./src/main/scala/HelloWorld.scala``::
+
+    package example
+
+    object Main {
+      def main(args: Array[String]): Unit =
+      println("Hello, world!")
+    }
+
+now simply run sbt run to get everything compiled and have the expected output!
```
Comment (andreaTP) [inline docs/user/sbt.rst]: sbt run need to be quoted ``` sbt run ```

File: docs/user/lang.rst [hunk 3]
```diff
@@ -0,0 +1,9 @@
+.. _lang:
+
+Language semantics
+==================
+
+In general, the semantics of the Scala Native language are the same as Scala on
```
Comment (andreaTP) [inline docs/user/lang.rst]: I think here is almost fair to mention memory allocation with ```stackalloc``` and ```malloc```.
And probably add a note on threading execution (i.e. not supported now)

**Comment by jonas** [comment]
Have you considered using [Paradox](https://github.com/lightbend/paradox) for Scala Native docs? It uses Markdown with a few custom directives for including source code etc. It is still a pretty new project and doesn't have all the features that Sphinx supports such as PDF generation, but in terms of build infra it would only require an SBT plugin and having "compile checked" code is pretty awesome.

**Comment by densh** [comment]
@jonas We have. Markdown dialects are very lock-in-prone and are painful to maintain long-term. At least that has been my experience with markdown docs on `docs.scala-lang.org` that I don't want to repeat again.

File: docs/user/lang.rst (hunk 3 above)
Comment (densh) [inline docs/user/lang.rst]: This pr only introduces basic layout. We'll fill it up with content as we go.

File: docs/user/sbt.rst (hunk 2 above)
Comment (densh) [inline docs/user/sbt.rst]: Fixed!

**Comment by densh** [comment]
@andreaTP Just about anything can be changed. Sphinx is extremely customisable.

**Pull request merged**